config/initializers/hostname.rb

```diff
@@ -1,16 +1,38 @@
-# Compute hostname and port from the hostname in the Settings
+# Compute hostname and port from the hostname in the Settings.
+# If no hostname is specified, read the hostname from the OS.
 module Hostname
   def self.fqdn
-    hostname = Settings.hostname.split(':').first
-    if hostname == 'localhost' || hostname.include?('.')
+    hostname = Settings.hostname.split(':').first if Settings.hostname
+    if hostname == 'localhost' || (hostname && hostname.include?('.'))
       hostname
     else
-      Addrinfo.tcp(tmp, 0).getnameinfo.first
+      begin
+        Addrinfo.tcp(Socket.gethostname, 0).getnameinfo.first
+      rescue ::SocketError => e
+        message = <<-MSG.strip_heredoc
+          Could not automatically determine the hostname:
+          #{e.class}: #{e.message}
+
+          Please set the hostname manually in the configuration
+          or consult the documentation of `gethostname`:
+           * http://ruby-doc.org/stdlib/libdoc/socket/rdoc/Socket.html#method-c-gethostname
+           * man gethostname
+           * http://linux.die.net/man/2/gethostname
+        MSG
+        $stderr.puts message
+        exit
+      end
     end
   end
 
   def self.port
-    Settings.hostname.split(':').last
+    Settings.hostname.split(':').last if Settings.hostname.include?(':')
+  end
+
+  def self.url_authority(scheme: 'http')
+    port = Ontohub::Application.config.port
+    port = ":#{port}" if port
+    "#{scheme}://#{Ontohub::Application.config.fqdn}#{port}/"
```

Comment:
The trailing slash should break every usage with loc-ids (which have a leading slash themselves).